src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -424,17 +468,28 @@ private void SchedulePollTask()
             Timers.StartSingleTimer(PollTimerKey, _pollMessage, _settings.PollInterval);
         }
 
-        private void CheckOverlappingRequests(string updateType, IActorRef fromStage, IImmutableSet<TopicPartition> topics)
+        private void RequestDelayedPoll()
+        {
+            if (_delayedPollInFlight)
+            {
+                _delayedPollInFlight = true;
+                Self.Tell(_delayedPollMessage);
+            }
+        }
+
+        private void CheckOverlappingRequests(string updateType, IActorRef fromStage,
+            IImmutableSet<TopicPartition> topics)
         {
             // check if same topics/partitions have already been requested by someone else,
             // which is an indication that something is wrong, but it might be alright when assignments change.
-            foreach (var (actorRef, request) in _requests.ToTuples())
```

Comment:
`.ToTuples` is generally not needed - you can just do deconstruction on all `KeyValuePair`s coming out of an `IDictionary`